### DISCUSSION TITLE
I may find the reason why open webui can't render latex well

### DISCUSSION THREAD

**Discussion by AeneasZhu** [open-webui/open-webui#2365] [General]
As is known, open webui supports latex rendering, but it can't render well when llm generates latex code, such as
![image](https://github.com/open-webui/open-webui/assets/44701446/df545e45-73de-4259-b415-36b21b995953)

As we can see, the two "$" is placed around the formula where the latex fails to be rendered. However, when I manually change the latex format by replacing "$" with "$$" around the latex, the render works.

![image](https://github.com/open-webui/open-webui/assets/44701446/5f1e35d3-ff29-450b-9830-c162b014b93e)
![image](https://github.com/open-webui/open-webui/assets/44701446/5df922e7-e51d-4e9d-9b88-fbe8710cdf38)

Therefore, unlike the latex renderer on VSCode markdown or nvim markdow, where latex can be rendered like `$latex formula$` or 
```
$$

latex formula

$$
```
it only supports format like `$$latex$$`. And when I ask open webui to generate formula with specific latex format like
![image](https://github.com/open-webui/open-webui/assets/44701446/d5a0cec0-d5e7-4ef1-a0fa-9e7a9dbdce84)

open webui did generate the latex format I wish for. And its original format is 
![image](https://github.com/open-webui/open-webui/assets/44701446/dde14c25-bc1f-4f6a-8cdf-c137a59fabb1)

where latex is placed around two "$$" and this is why I find out the missing point that open webui can't render latex as we wish for.

In the end, could  there be any improvement for this?

**Comment by alexkermani** [comment]
Seconding this issue, I found this thread looking for others with un-rendered LaTeX.

Realized I have a different delimiter `\[  LaTeX \]` 
Some of my lines didn't render because the response didn't have that delimiter:
## Raw response missing `\[  LaTeX \]` 
--- 
![Screenshot 2024-05-22 at 19 12 56](https://github.com/open-webui/open-webui/assets/13397426/b12de753-c5be-4cef-9b53-486480827f22)

## Raw response with `\[  LaTeX \]` :
---
![Screenshot 2024-05-22 at 19 06 35](https://github.com/open-webui/open-webui/assets/13397426/5859c686-5056-4b58-b703-4adaa6bab463)
---
## Which renders as:

![Screenshot 2024-05-22 at 19 07 14](https://github.com/open-webui/open-webui/assets/13397426/9269b0fe-d809-4dcc-aec0-1581d49c653f)
---

When I remove the dot character that cdot error goes away
